### DISCUSSION TITLE
Oneliner to convert Markdown to PDF

### DISCUSSION THREAD

**Discussion by ondics** [gotenberg/gotenberg#632] [Q&A]
I'm looking for a oneliner using Gotenberg Docker image to convert a single markdown file to a pdf in batch mode. 

Something like this...

    docker run --rm -p 3000:3000 thecodingmachine/gotenberg:7 --inputfile=hello.md --converter=markdown-to-pdf

Is this possible?

**Comment by gulien** [comment]
Hello @ondics,

Not it's not possible; Gotenberg only works as an API. 😄

**Reply by ondics** [reply]
Don't want to give up already...

Maybe can write something as

    cat hello.md | docker run [...] --outputfile=hello.pdf unoconv ...

and overwrite the command to the entrypoint?

Don't know if this is working.

Or another way: using curl in the container. And posting the mounted file to Gotenberg?

    docker run -v $PWD:/mydir --rm thecodingmachine/gotenberg:7 curl -x POST http://localhost:3000 -d @/mydir/hello.md -o hello.pdf

**Reply by gulien** [reply]
I mean you could make it work like that, but a better option would be to write a script that:

1. Starts a Gotenberg instance in background mode
2. Does a curl request
3. Stop the Gotenberg instance

**Reply by ondics** [reply]
A oneliner is better from my point of view since it does not depend on the scripting environment nor requires a script at all. 

That's why I want to do the processing inside the container.
